src/bun_utils.c

```diff
@@ -129,15 +129,18 @@ open_real_file(const char *name)
 	    S_IRUSR | S_IWUSR);
 	if (fd < 0)
 		goto error;
```

Comment:
We will go-to error before we try the `open`. 
```suggestion
		fd = open(final_name, O_CREAT | O_CLOEXEC | O_TRUNC | O_RDWR,
		    S_IRUSR | S_IWUSR);
		if (fd < 0)
			goto error;
		if (unlink(final_name) == -1)
			goto error;
	}
```